### PR TITLE
Feature/reducers and actions

### PR DESCRIPTION
**Pull request by Braingenius2**
# Added reducers and actions

This pull request adds Redux to the bookstore application. Redux is a state management library that helps to keep track of the application state. 

In this pull request, I implemented the following:

- [x] Installed Redux Toolkit and React Redux
- [x] Created a directory to store Redux logic
- [x] Configured a Redux store
- [x] Defined a slice of state for books
- [x] Defined a slice of state for categories
- [x] Structured the application files using a "feature folder" approach and uses the ducks pattern

### CODE REVIEW

File: src/redux/store.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { configureStore } from '@reduxjs/toolkit';
+
+const store = configureStore({
+  reducer: {},
+});
```
Comment (brainconnect93) [inline src/redux/store.js:10]: - [ ] I would kindly suggest you import the books and categories as a Reducer in your store.

N.B: 
- Imports books from src/redux/books.
- Imports categories from src/redux/categories.
- Uses configureStore to create a reducer.
     - books and categories should be added to configureStore method.


Suggested solution :point_down: 
```JS
// ./src/redux/store.js

import { configureStore } from "@reduxjs/toolkit"
import booksReducer from "./books/booksSlice"
import categoriesReducer from "./categories/categoriesSlice"

const store = configureStore({
  reducer: { books: booksReducer, categories: categoriesReducer },
})

export default store
```

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = 'Under construction';
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {},
+});
```
Comment (brainconnect93) [inline src/redux/categories/categoriesSlice.js:15]: - [ ] According to the project requirement, it was stated that our categories should Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

Suggested solution :point_down:
```JS
// ./src/redux/categories/categoriesSlice.js

import { createSlice } from "@reduxjs/toolkit"

export const categoriesSlice = createSlice({
  name: "categories",
  initialState: {
    categories: [],
  },
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
})

export const { checkStatus } = categoriesSlice.actions

export default categoriesSlice.reducer
```

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @Braingenius2 , 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To Highlight :1st_place_medal: 

- Descriptive Pull Request. :heavy_check_mark:
- Action and Reducer functionality. :heavy_check_mark:
- Ducks pattern file structure. :heavy_check_mark:
- `@reduxjs/toolkit` & `react-redux` present in package.json :heavy_check_mark:
- Linters environment properly setup :heavy_check_mark:
- Good use of Git flow :heavy_check_mark:

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 2 above)
Comment (Braingenius2) [reply src/redux/categories/categoriesSlice.js:15]: Thank you for taking the time to review my code. I really appreciate the valuable insights you provided 🙇 . I'll make sure to incorporate your recommendations promptly.

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @Braingenius2 

Great work implementing the suggested changes 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Redux Store is configured, perfect!
-  ✅ Great work using the right GitFlow 
-  ✅ Your README looks good, well done!
-  ✅ Linters are working properly, great job! 

## Optional suggestions ❕

- [OPTIONAL] The issues page link in **_Contributing_** section is redirecting to Math Magician's issues page ⚠ Kindly fix this to avoid any confusion for the users who wants to interact with you📌

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**